app/models/spreadsheets/primary_school_spreadsheet.rb

```diff
@@ -16,31 +16,29 @@ def import(file_path)
     end
   end
 
+  def process(row)
+    commune = get_commune(row)
+    return if commune.blank?
+
+    school = ::PrimarySchool.find_or_initialize_by(code: row["school_code"].to_s.strip)
+    school.code = school.code.presence || build_school_code(commune)
+    school.name_km = row["school_name_km"]
+    school.name_en = row["school_name_en"] || row["school_name_km"]
+    school.commune_id ||= commune.id
+    school.district_id ||= commune.district_id
+    school.province_id ||= commune.province_id
+    school.save
+  end
+
   private
     def spreadsheet(file_path)
       Roo::Spreadsheet.open(file_path)
     end
 
-    def process(row)
-      commune = get_commune(row)
-      return if commune.blank?
-
-      school = ::PrimarySchool.find_or_initialize_by(code: row["code"].strip)
-      school.code ||= build_school_code(commune)
-      school.name_km = row["name_km"]
-      school.name_en = row["name_en"] || row["name_km"]
-      school.commune_id ||= commune.id
-      school.district_id ||= commune.district_id
-      school.province_id ||= commune.province_id
-      school.save
-    end
-
     def get_commune(row)
-      return if row["commune"].blank?
+      return if row["commune_code"].blank?
```

Comment:
@hengsokly this method should return commune object or nil object rather than just return so from the calling point please check with .nil?